tests/container-files/etc/vsftpd/vsftpd.conf

```diff
@@ -0,0 +1,75 @@
+# Run in the foreground to keep the container running:
+background=NO
+
+# Allow anonymous FTP? (Beware - allowed by default if you comment this out).
+anonymous_enable=NO
+
+# Uncomment this to allow local users to log in.
+local_enable=YES
+
+## Enable virtual users
+guest_enable=YES
+
+## Virtual users will use the same permissions as anonymous
+virtual_use_local_privs=YES
+
+# Uncomment this to enable any form of FTP write command.
+write_enable=YES
+
+## PAM file name
+pam_service_name=vsftpd_virtual
+
+## Home Directory for virtual users
+user_sub_token=$USER
+local_root=/home/vsftpd/$USER
+
+# You may specify an explicit list of local users to chroot() to their home
+# directory. If chroot_local_user is YES, then this list becomes a list of
+# users to NOT chroot().
+chroot_local_user=YES
+
+# Workaround chroot check.
+# See https://www.benscobie.com/fixing-500-oops-vsftpd-refusing-to-run-with-writable-root-inside-chroot/
+# and http://serverfault.com/questions/362619/why-is-the-chroot-local-user-of-vsftpd-insecure
+allow_writeable_chroot=YES
+
+## Hide ids from user
+hide_ids=YES
+
+## Passive Address that gets advertised by vsftpd when responding to PASV command
+pasv_address=
+
+## Enable passive mode
+pasv_enable=YES
+
+## Set passive port range
+pasv_max_port=50000
+pasv_min_port=40000
+
+## Enable logging
+xferlog_enable=YES
+vsftpd_log_file=/var/log/vsftpd/vsftpd.log
+
+## Enable active mode
+port_enable=YES
+connect_from_port_20=YES
+ftp_data_port=20
+
+## control umask of uploaded files
+# * 077 means that uploaded files get rw- --- ---
+# * 022 means that uploaded files get rw- r-- r--
+local_umask=022
+
+ssl_enable=YES
+allow_anon_ssl=NO
+force_local_data_ssl=NO
+force_local_logins_ssl=NO
+ssl_tlsv1_1=YES
+ssl_tlsv1_2=YES
+ssl_tlsv1=NO
+ssl_sslv2=NO
+ssl_sslv3=NO
+require_ssl_reuse=YES
+ssl_ciphers=HIGH
+rsa_cert_file=/etc/ssl/certs/vsftpd.crt
+rsa_private_key_file=/etc/ssl/private/vsftpd.key
```

Comment:
This is the `vsftpd` server setting for the fake FTP server container.

Reply:
I suggest to allows using the anonymous mode, because in the feature the library may support this mode like adding a new connection class like an `FtpAnonymousConnection` class.

Reply:
The anonymous mode setting is enabled in the latest commit.